src/utils/object.js

```diff
@@ -143,3 +145,34 @@ export function omit(o, props) {
     return acc;
   }, {});
 }
+
+/**
+ * @param {!JsonObject} unusedTarget
+ * @param {...(!JsonObject|null|undefined)} var_args
+ * @return {!JsonObject}
+ */
+export function jsonObjectAssign(unusedTarget, var_args) {
+  return /** @type {!JsonObject} */ (Object.assign.apply(null, arguments));
+}
+
+/**
+ * @param {!JsonObject} target
+ * @param {!JsonObject} source
+ * @param {number} depth the maximum merge depth. if exceeded, object.assign
+ *                       will be used instead.
+ * @return {!JsonObject}
+ * @throws {error} if source contains a circular reference.
+ * Note: only nested objects are deep-merged, primitives and arrays are not.
+ */
+export function jsonObjectDeepMerge(target, source, depth) {
+  return /** @type {!JsonObject} */ (deepMerge(target, source, depth));
+}
+
+/**
+ * @param {!JsonObject} o An json object to remove properties from
+ * @param {!Array<string>} props A list of properties to remove from the Object
+ * @return {!JsonObject} An json object with the given properties removed
+ */
+export function jsonObjectOmit(o, props) {
+  return /** @type {!JsonObject} */ (omit(o, props));
+}
```

Comment:
Minor: Would be a bit nicer if we could make these static functions similar to `Object.assign`.

```js
JsonObject.assign(...);

Object.assignJson(...);
```